### ISSUE TITLE
"An input couldnot be retrieved. It could be because a worker has died." Error is popping up when training model

### ISSUE THREAD

**Issue by SuganyaMurugesan** [matterport/Mask_RCNN#1873]
Hi,

I am using Mask RCNN to train data on traffic lights.
When i tried to train the model. It is showing 

"Epoch 1/30 /usr/local/lib/python3.6/dist-packages/keras/utils/data_utils.py:709: UserWarning: An input could not be retrieved. It could be because a worker has died.We do not have any information on the lost sample. UserWarning)"

I am not sure on which input it is waiting for.
I have modified the dataset load functionalities as below to accomodate on my data.

`lass TrafficLightDataset(utils.Dataset):

    def load_trafficlight(self, dataset_dir, subset):
        """Load a subset of the Balloon dataset.
        dataset_dir: Root directory of the dataset.
        subset: Subset to load: train or val
        """
        # Add classes. We have only one class to add.
        self.add_class("trafficlight", 1, "trafficlight")

        # Train or validation dataset?
        assert subset in ["train", "val"]
        dataset_dir = os.path.join(dataset_dir, subset)

        # Load annotations
        # VGG Image Annotator (up to version 1.6) saves each image in the form:
        # { 'filename': '28503151_5b5b7ec140_b.jpg',
        #   'regions': {
        #       '0': {
        #           'region_attributes': {},
        #           'shape_attributes': {
        #               'all_points_x': [...],
        #               'all_points_y': [...],
        #               'name': 'polygon'}},
        #       ... more regions ...
        #   },
        #   'size': 100202
        # }
        # We mostly care about the x and y coordinates of each region
        # Note: In VIA 2.0, regions was changed from a dict to a list.
        annotations = json.load(open(os.path.join(dataset_dir, "annotate_region.json")))
        
        # Add images
        for a in annotations:
          rect = a["boxes"]
          image_id = a["path"].split('/')[4]
          path = "data/"+ image_id
          image_path = os.path.join(dataset_dir, path)
          image = skimage.io.imread(image_path)
          height, width = image.shape[:2]
          self.add_image(
                "trafficlight",
                image_id=image_id,  # use file name as a unique image id
                path=image_path,
                width=width, height=height,
                rect=rect)

    def load_mask(self, image_id):
        """Generate instance masks for an image.
       Returns:
        masks: A bool array of shape [height, width, instance count] with
            one mask per instance.
        class_ids: a 1D array of class IDs of the instance masks.
        """
        # If not a balloon dataset image, delegate to parent class.
        image_info = self.image_info[image_id]
        if image_info["source"] != "trafficlight":
            return super(self.__class__, self).load_mask(image_id)

        # Convert polygons to a bitmap mask of shape
        # [height, width, instance_count]
        info = self.image_info[image_id]
        mask = np.zeros([info["height"], info["width"], len(info["rect"])],
                        dtype=np.uint8)
        for i, p in enumerate(info["rect"]):
            # Get indexes of pixels inside the polygon and set them to 1
            start = (p["x_min"], p["y_min"])
            end = (p["x_max"], p["y_max"])
            rr, cc = skimage.draw.polygon(start, end)
            mask[rr, cc, i] = 1

        # Return mask, and array of class IDs of each instance. Since we have
        # one class ID only, we return an array of 1s
        return mask.astype(np.bool), np.ones([mask.shape[-1]], dtype=np.int32)

    def image_reference(self, image_id):
        """Return the path of the image."""
        info = self.image_info[image_id]
        if info["source"] == "trafficlight":
            return info["path"]
        else:
            super(self.__class__, self).image_reference(image_id)
`
If someone has faced this issue. can you please help out.

**Comment by osvadimos** [comment]
Faced same issue.

**Comment by SuganyaMurugesan** [comment]
@osvadimos : 

I faced this issue because i mistakenly drawn polygon in place of rectangle when feeding the model with the masks. I was successfully able to train once i have drawn polygon in place of rectangle.
Running the data inspection will help you narrow down on the problem before starting to train the model. Hope it helps.
Happy coding.

**Comment by osvadimos** [comment]
> @osvadimos :
> 
> I faced this issue because i mistakenly drawn polygon in place of rectangle when feeding the model with the masks. I was successfully able to train once i have drawn polygon in place of rectangle.
> Running the data inspection will help you narrow down on the problem before starting to train the model. Hope it helps.
> Happy coding.

Thanks. Partly that was the issue. You've saved my day.

**Comment by leventozdemir** [comment]
> @osvadimos :
> 
> I faced this issue because i mistakenly drawn polygon in place of rectangle when feeding the model with the masks. I was successfully able to train once i have drawn polygon in place of rectangle.
> Running the data inspection will help you narrow down on the problem before starting to train the model. Hope it helps.
> Happy coding.

i have the same problem only when i'm trying to train on 3 classes 
{
    def image_reference(self, image_id):
        """Return the path of the image."""
        info = self.image_info[image_id]
        if info["source"] == "right":
            return info["path"]
        elif info["source"] == "left":
            return info["path"]
        else:
            super(self.__class__, self).image_reference(image_id)
}
{
            self.add_image(
                "right",  ## for a single class just add the name here
                image_id=a['filename'],  # use file name as a unique image id
                path=image_path,
                width=width, height=height,
                polygons=polygons)
            self.add_image(
                "left",  ## for a single class just add the name here
                image_id=a['filename'],  # use file name as a unique image id
                path=image_path,
                width=width, height=height,
                polygons=polygons)
}
{
NAME = "leg"

    # We use a GPU with 12GB memory, which can fit two images.
    # Adjust down if you use a smaller GPU.
    IMAGES_PER_GPU = 1

    # Number of classes (including background)
    NUM_CLASSES = 2 + 1  # Background + toy

    # Number of training steps per epoch
    STEPS_PER_EPOCH = 100

    # Skip detections with < 90% confidence
    DETECTION_MIN_CONFIDENCE = 0.9

}
i made this 3 changes on the code 
what must i do  ?

**Comment by MSOHAPPY** [comment]
Facing the same issue, I am running the offical balloon code, how to solve this issue
